kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -186,8 +196,15 @@ public void write() {
     while (!buffer.isEmpty()) {
       try {
         executeState(now);
-      } catch (SchemaProjectorException | IllegalWorkerStateException e) {
+      } catch (IllegalWorkerStateException e) {
         throw new ConnectException(e);
+      } catch (SchemaProjectorException e) {
+        if (reporter != null) {
+          reporter.report(buffer.poll(), e);
+          log.warn("Errant record written to DLQ due to: {}", e.getMessage());
+        } else {
+          throw e;
+        }
```

Comment:
do we want to skip the record if we report it

Reply:
We'd want to skip writing the record to the S3 file if it's been reported, is that what you mean?

Reply:
as long as customers understand that enabling the reporter implicitly subscribes them to skipping records instead of failing